api/controllers/OpenEAppController.js

```diff
@@ -90,20 +93,31 @@ const OpenEAppController = {
         if (!casebookResponse) {
             throw new Error('No data received from Casebook');
         }
+        if (!casebookResponse.documents) {
+            throw new Error('No documents found from Casebook');
+        }
+        if (!casebookResponse.payment) {
+            throw new Error('No payment info found from Casebook');
+        }
+
         return {
             applicationId: applicationTableData.unique_app_id,
             dateSubmitted: OpenEAppController._formatDate(
                 applicationTableData.createdAt
             ),
             documents: casebookResponse.documents,
             originalCost: HelperService.formatToUKCurrency(
-                casebookResponse.payment.transactions[0].amount
+                casebookResponse.payment.transactions[0].amount || 0
```

Comment:
I see you've added the guard condition above for when payment is not specified but you probably need the same for when payment.transactions are not specified as well - not sure if that can even happen but worth checking

Reply:
Yeah, it makes sense. Max told me about this library (https://github.com/gcanti/io-ts) which makes sure the response from the API matches a specific object and if it doesn't, it responds with a failure. Unfrountatley this app wasn't written in TypeScript but if it was then this library would be very useful.